### DISCUSSION TITLE
Does topicDataType include module prefix? [16141]

### DISCUSSION THREAD

**Discussion by srad0** [eProsima/Fast-DDS#3081] [Q&A] [answered]
I'm following the HelloWorld example and unable to get my pub/sub talking. One question I have is, I have an IDL defined as follows.

```
module myMod {

struct MyStruct
{
    string myString;
};

};

```
And I have a static xml such as:

```
<staticdiscovery>
    <participant>
        <name>myTest</name>
        <reader>
            <entityId>1</entityId>
            <userId>2</userId>
            <topicName>MyTopic</topicName>
            <topicDataType>MyStruct</topicDataType>
        </reader>
    </participant>
</staticdiscovery>
```

Should the `topicDataType` be `MyStruct` or include the module prefix as well and be `myMod::MyStruct` in the XML file?

**Comment by richiware** [comment, marked as answer]
Full type name is expected, including the namespaces. In your case, the full type name of your type is `myMod::MyStruct`.
